message_converter/src/converter_node.cpp

```diff
@@ -1,93 +1,32 @@
-        Matrix3x3d ret;
-        ret(0, 0) = -sLat * cLon;
-        ret(0, 1) = -sLat * sLon;
-        ret(0, 2) = cLat;
-        ret(1, 0) = -sLon;
-        ret(1, 1) = cLon;
-        ret(1, 2) = 0.0;
-        ret(2, 0) = cLat * cLon;
-        ret(2, 1) = cLat * sLon;
-        ret(2, 2) = sLat;
-        return ret;
-    }
-};
+#include <converter_node.hpp>
 
 SwiftNavRover::SwiftNavRover(ros::NodeHandle* node_handle, double hla = 0.0, double hlo = 0.0, double hal = 0.0):nh(*node_handle), 
-home_lat(hla), home_lon(hlo), home_alt(hal){
+home_lat_(hla), home_lon_(hlo), home_alt_(hal){
     SwiftNavRover::init();
-    SwiftNavRover::home_lon_rad = SwiftNavRover::deg2rad(home_lon);
-    SwiftNavRover::home_lat_rad = SwiftNavRover::deg2rad(home_lat);
+    SwiftNavRover::home_lon_rad_ = SwiftNavRover::deg2rad(home_lon_);
+    SwiftNavRover::home_lat_rad_ = SwiftNavRover::deg2rad(home_lat_);
     double x,y,z;
-    SwiftNavRover::geodetic2ecef(home_lat, home_lon, home_alt, &x, &y, &z);
+    SwiftNavRover::geodetic2ecef(home_lat_, home_lon_, home_alt_, &x, &y, &z);
     double phip = atan2(z, sqrt(pow(x,2)+pow(y,2)));
-    SwiftNavRover::ecef_to_ned_matrix = SwiftNavRover::nRe(phip, SwiftNavRover::home_lon_rad);
+    SwiftNavRover::ecef_to_ned_matrix = SwiftNavRover::nRe(phip, SwiftNavRover::home_lon_rad_);
 }
 
 void SwiftNavRover::init()
 {
-    vel_sub = nh.subscribe("/rover/piksi/position_receiver_0/sbp/vel_ned_cov", 1000, &SwiftNavRover::baselineVelocityCallback, this);
-    pos_sub = nh.subscribe("/rover/piksi/position_receiver_0/sbp/baseline_ned", 1000, &SwiftNavRover::baselinePositionCallback, this);
-    pos_llh_sub = nh.subscribe("/rover/piksi/position_receiver_0/sbp/pos_llh", 1000, &SwiftNavRover::posLLHCallback, this);
-    ned_point_fix = nh.advertise<geometry_msgs::PointStamped>("/rover/ned_point_fix", 1000);
-    ned_baseline_position_fix = nh.advertise<gnss_msgs::BaselinePosition>("/rover/ned_baseline_position_fix", 1000);
-    ned_vel_cov_fix = nh.advertise<gnss_msgs::BaselineVelocity>("/rover/ned_vel_cov_fix", 1000);
+    // Subscribers
+    vel_sub_ = nh.subscribe("/rover/piksi/position_receiver_0/sbp/vel_ned_cov", 1000, &SwiftNavRover::baselineVelocityCallback, this);
+    pos_sub_ = nh.subscribe("/rover/piksi/position_receiver_0/sbp/baseline_ned", 1000, &SwiftNavRover::baselinePositionCallback, this);
+    pos_llh_sub_ = nh.subscribe("/rover/piksi/position_receiver_0/sbp/pos_llh", 1000, &SwiftNavRover::posLLHCallback, this);
+
+    // Publishers
+    ned_point_fix_ = nh.advertise<geometry_msgs::PointStamped>("/rover/ned_point_fix", 10);
+    ned_baseline_position_fix_ = nh.advertise<gnss_msgs::BaselinePosition>("/rover/ned_baseline_position_fix", 10);
+    ned_vel_cov_fix_ = nh.advertise<gnss_msgs::BaselineVelocity>("/rover/ned_vel_cov_fix", 10);
+    rtk_mode_available_ = nh.advertise<std_msgs::Bool>("/rover/rtk_available", 10);
+    diagnostic_publisher_ = nh.advertise<diagnostic_msgs::DiagnosticArray>("/diagnostics_rover", 10);
```

Comment:
Will change this topic to ```/diagnostics```. It was set to ```/diagnostics_rover``` for some debugging

Reply:
Fixed in https://github.com/Humatics/ethz_piksi_ros/pull/1/commits/398f00acf0b225459f23393d5c03329c25f94e21